deepfence_server/model/registry.go

```diff
@@ -622,33 +663,27 @@ func RegistrySummary(ctx context.Context, registryID mo.Option[string], registry
 	defer tx.Close()
 
 	queryPerRegistry := `
-	MATCH (n:RegistryAccount{node_id:$id})-[:HOSTS]->(m:ContainerImage)-[:IS]-(i:ImageStub)
-	WITH n,
-		COUNT(distinct m.docker_image_name) AS images,
-		COLLECT(distinct i.tags) AS tags,
-		COUNT(distinct n) AS registries
+    MATCH (n:RegistryAccount{node_id:$id})
+    WITH COUNT(distinct n) as registries
+	MATCH (n:RegistryAccount{node_id:$id})-[:HOSTS]->(i:ImageStub)-[:IS]-(m:ContainerImage)
+	MATCH (n)-[:HOSTS]->(m)
+	WITH registries, n,
+		COUNT(distinct i.docker_image_name) AS repositories,
+		COUNT(distinct m.docker_image_id) AS images
 	OPTIONAL MATCH (s)-[:SCANNED]->()<-[:HOSTS]-(n)
-	RETURN COLLECT(s.status) AS scan_status, images, reduce(total = 0, t IN tags | total + size(t)) as tags, registries
-	`
-
-	queryRegistriesByType := `
-	MATCH (n:RegistryAccount{registry_type:$type})-[:HOSTS]->(m:ContainerImage)-[:IS]-(i:ImageStub)
-	WITH
-		COUNT(distinct m.docker_image_name) AS images,
-		COLLECT(distinct i.tags) AS tags,
-		COUNT(distinct n) AS registries
-	OPTIONAL MATCH (s)-[:SCANNED]->()<-[:HOSTS]-(:RegistryAccount{registry_type:$type})
-	RETURN COLLECT(s.status) AS scan_status, images, reduce(total = 0, t IN tags | total + size(t)) as tags, registries
+	RETURN COLLECT(s.status) AS scan_status, repositories, images, registries
 	`
 
 	queryAllRegistries := `
-	MATCH (n:RegistryAccount)-[:HOSTS]->(m:ContainerImage)-[:IS]-(i:ImageStub)
-	WITH
-		COUNT(distinct m.docker_image_name) AS images,
-		COLLECT(distinct i.tags) AS tags,
-		COUNT(distinct n) AS registries
+	MATCH (n:RegistryAccount)
+	WITH COUNT (distinct n) as registries
+	MATCH (n:RegistryAccount)-[:HOSTS]->(i:ImageStub)-[:IS]-(m:ContainerImage)
+	MATCH (n)-[:HOSTS]->(m)
+	WITH registries,
+		COUNT(distinct i.docker_image_name) AS repositories,
+		COUNT(distinct m.docker_image_id) AS images
 	OPTIONAL MATCH (s)-[:SCANNED]->()<-[:HOSTS]-(:RegistryAccount)
-	RETURN COLLECT(s.status) AS scan_status, images, reduce(total = 0, t IN tags | total + size(t)) as tags, registries
+	RETURN COLLECT(s.status) AS scan_status, registries, repositories, images
```

Comment:
Let's make this query in 2:
One to retrieve the `registries` count
And the rest

Reply:
Done.